pypdf/_utils.py

```diff
@@ -347,14 +347,11 @@ def b_(s: Union[str, bytes]) -> bytes:
         return bc[s]
     try:
         r = s.encode("latin-1")
-        if len(s) < 2:
-            bc[s] = r
-        return r
     except Exception:
```

Comment:
With the latest change, isn't this exception too broad?

Reply:
The exception is now:
```
try:
    r = s.encode("latin-1")
except Exception:
    r = s.encode("utf-8")
```
The caching
```
if len(s) < 2:
    bc[s] = r
return r

```
 is now outside this as it was previously duplicated on both sides of the exception.

We use b_ in quite a few places and renaming it bytes_ would be good as it is more self-explanatory.

Reply:
But which exceptions should be raised now except an UnicodeDecodeError which we could catch? The comparison as before might raise other exceptions as well.

Reply:
If the less than comparison does raise an exception in the try block it would still raise an exception when it got to the exception block, unless running `r = s.encode("utf-8") would prevent this?

Reply:
This is not really what I am arguing. If we attempt to optimize the code, `Exception` is a too broad exception class with the new variant in my opinion, as there barely are exceptions which could be raised here. Thus I am asking to replace the exception handling with the specific exception which actually can occur here.

Reply:
Good idea. Have replaced, now is a UnicodeEncodeError.

Reply:
Renaming `b_` to `bytes_` would be more consistent with str_ and is self-documenting.

@pubpub-zz and @stefan6419846 thoughts?

Reply:
Would renaming be a breaking change?

Reply:
I did the rename #2791; it passes our tests.